crates/core/src/preview1.rs

```diff
@@ -0,0 +1,193 @@
+use std::{any::Any, path::PathBuf};
```

Comment:
Would it make sense to write a module doc explaining what these implementations are for?